<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot
  name="sanbot_nano_urdf">
  <link
    name="base_link">
    <inertial>
      <origin
        xyz="1.64477253035502E-05 -5.93128846061813E-06 0.0729321766717518"
        rpy="0 0 0" />
      <mass
        value="6.0" />
      <inertia
        ixx="0.11281645397611"
        ixy="-2.49617328764965E-05"
        ixz="3.29336671339688E-05"
        iyy="0.11397342921281"
        iyz="-5.30882435321482E-07"
        izz="0.213909779379668" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://sanbot_nano_urdf/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://sanbot_nano_urdf/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="wheel_back_link">
    <inertial>
      <origin
        xyz="-1.95717879520655E-08 1.68817821135508E-07 0.0179686472612415"
        rpy="0 0 0" />
      <mass
        value="0.088" />
      <inertia
        ixx="9.20122787951246E-05"
        ixy="-3.94042367835364E-16"
        ixz="-1.19493108152976E-20"
        iyy="9.20122787946692E-05"
        iyz="3.93901867754232E-21"
        izz="0.000170504431903692" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://sanbot_nano_urdf/meshes/wheel_back_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.101960784313725 0.101960784313725 0.101960784313725 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://sanbot_nano_urdf/meshes/wheel_back_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="wheel_back"
    type="continuous">
    <origin
      xyz="-0.1195 0 0.03"
      rpy="1.5707963267949 0 -1.5707963267949" />
    <parent
      link="base_link" />
    <child
      link="wheel_back_link" />
    <axis
      xyz="0 0 1" />
  </joint>
  <link
    name="wheel_left_link">
    <inertial>
      <origin
        xyz="-0.0155613556791648 -1.36414627760484E-07 -0.00898423585151892"
        rpy="0 0 0" />
      <mass
        value="0.088" />
      <inertia
        ixx="0.000150881393626606"
        ixy="-4.80725859037333E-20"
        ixz="3.39880992947803E-05"
        iyy="9.20122787944416E-05"
        iyz="7.32731581959123E-20"
        izz="0.000111635317072437" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://sanbot_nano_urdf/meshes/wheel_left_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.101960784313725 0.101960784313725 0.101960784313725 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://sanbot_nano_urdf/meshes/wheel_left_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="wheel_left"
    type="continuous">
    <origin
      xyz="0.0597500000000005 0.10349003575224 0.03"
      rpy="1.5707963267949 0 -1.5707963267949" />
    <parent
      link="base_link" />
    <child
      link="wheel_left_link" />
    <axis
      xyz="-0.866025403784436 0 -0.500000000000004" />
  </joint>
  <link
    name="wheel_right_link">
    <inertial>
      <origin
        xyz="0.0155629154939929 2.62956446860085E-06 -0.00898153417298636"
        rpy="0 0 0" />
      <mass
        value="0.088" />
      <inertia
        ixx="0.000150881393626607"
        ixy="-1.12621292193358E-19"
        ixz="-3.39880992947797E-05"
        iyy="9.20122787944418E-05"
        iyz="-1.85546757553949E-19"
        izz="0.000111635317072436" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://sanbot_nano_urdf/meshes/wheel_right_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.101960784313725 0.101960784313725 0.101960784313725 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://sanbot_nano_urdf/meshes/wheel_right_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="wheel_right"
    type="continuous">
    <origin
      xyz="0.0597527016785317 -0.103488475937413 0.0299975068501592"
      rpy="1.5707963267949 0 -1.5707963267949" />
    <parent
      link="base_link" />
    <child
      link="wheel_right_link" />
    <axis
      xyz="0.866025403784443 0 -0.499999999999992" />
  </joint>
  <link
    name="body_link">
    <inertial>
      <origin
        xyz="-6.50746287136447E-07 0.171161303173797 0.00531466947071041"
        rpy="0 0 0" />
      <mass
        value="10.54" />
      <inertia
        ixx="0.626323286825396"
        ixy="4.13879140447484E-06"
        ixz="1.17478367883931E-07"
        iyy="0.583524845241408"
        iyz="-0.000761556134756353"
        izz="0.656661935026136" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://sanbot_nano_urdf/meshes/body_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://sanbot_nano_urdf/meshes/body_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="body"
    type="fixed">
    <origin
      xyz="0 0 0.14"
      rpy="1.5707963267949 0 -1.5707963267949" />
    <parent
      link="base_link" />
    <child
      link="body_link" />
    <axis
      xyz="0 0 0" />
  </joint>
  <link
    name="wing_left_link">
    <inertial>
      <origin
        xyz="-0.0180528514790016 -0.060356471352447 -0.00920090040634635"
        rpy="0 0 0" />
      <mass
        value="0.09" />
      <inertia
        ixx="0.00143508347758134"
        ixy="-1.11397432746257E-05"
        ixz="-1.72108088017256E-06"
        iyy="0.000410412505557005"
        iyz="-8.69465392904125E-05"
        izz="0.00103848643600513" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://sanbot_nano_urdf/meshes/wing_left_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://sanbot_nano_urdf/meshes/wing_left_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="wing_left"
    type="revolute">
    <origin
      xyz="-0.171 0.337999997444124 -0.00876845650366815"
      rpy="0 0 0" />
    <parent
      link="body_link" />
    <child
      link="wing_left_link" />
    <axis
      xyz="1 0 0" />
    <limit
      lower="-1.5708"
      upper="3.1416"
      effort="10"
      velocity="1" />
  </joint>
  <link
    name="wing_right_link">
    <inertial>
      <origin
        xyz="0.0180596331580364 -0.060389654065786 -0.00920199725760526"
        rpy="0 0 0" />
      <mass
        value="0.09" />
      <inertia
        ixx="0.00143527781768375"
        ixy="1.11307475747432E-05"
        ixz="1.71994140484216E-06"
        iyy="0.000410538848951017"
        iyz="-8.69715731520652E-05"
        izz="0.00103855929783202" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://sanbot_nano_urdf/meshes/wing_right_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://sanbot_nano_urdf/meshes/wing_right_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="wing_right"
    type="revolute">
    <origin
      xyz="0.171 0.337999997444124 -0.00876845650366815"
      rpy="0 0 0" />
    <parent
      link="body_link" />
    <child
      link="wing_right_link" />
    <axis
      xyz="1 0 0" />
    <limit
      lower="-1.5708"
      upper="3.1416"
      effort="10"
      velocity="1" />
  </joint>
  <link
    name="neck_link">
    <inertial>
      <origin
        xyz="5.07959831122588E-08 0.0941822883771556 -2.15444560226558E-07"
        rpy="0 0 0" />
      <mass
        value="0.014" />
      <inertia
        ixx="0.00036601965027074"
        ixy="2.06341914062177E-10"
        ixz="2.67552661174392E-10"
        iyy="2.63393480841545E-05"
        iyz="1.44182677610015E-09"
        izz="0.000376086984549767" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://sanbot_nano_urdf/meshes/neck_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://sanbot_nano_urdf/meshes/neck_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="neck"
    type="revolute">
    <origin
      xyz="0 0.37832467677926 0"
      rpy="0 0 0" />
    <parent
      link="body_link" />
    <child
      link="neck_link" />
    <axis
      xyz="0 -1 0" />
    <limit
      lower="-1.5708"
      upper="1.5708"
      effort="2"
      velocity="1" />
    <dynamics
      damping="0.1"
      friction="0.1" />
  </joint>
  <link
    name="head_link">
    <inertial>
      <origin
        xyz="-0.0029281136953756 -8.79951278457592E-06 -0.0046759697686755"
        rpy="0 0 0" />
      <mass
        value="2.0" />
      <inertia
        ixx="0.157224047016489"
        ixy="-9.02313066752037E-08"
        ixz="0.0118768228744953"
        iyy="0.152781569466689"
        iyz="-2.96042696367125E-06"
        izz="0.158625545969497" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://sanbot_nano_urdf/meshes/head_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://sanbot_nano_urdf/meshes/head_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="head"
    type="revolute">
    <origin
      xyz="0 0.165 0"
      rpy="0 0 1.5707963267949" />
    <parent
      link="neck_link" />
    <child
      link="head_link" />
    <axis
      xyz="0 -1 0" />
    <limit
      lower="0"
      upper="0.656"
      effort="2"
      velocity="1" />
    <dynamics
      damping="0.1"
      friction="0.1" />
  </joint>
  <link
    name="camera_chin_link">
    <inertial>
      <origin
        xyz="3.44038304961544E-17 1.11022302462516E-16 0.0012729648678558"
        rpy="0 0 0" />
      <mass
        value="0.002" />
      <inertia
        ixx="2.75893420315291E-10"
        ixy="-9.07141086890545E-27"
        ixz="-2.91066093183747E-27"
        iyy="3.49794254771563E-10"
        iyz="-1.32958464693283E-26"
        izz="5.62491899186398E-10" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://sanbot_nano_urdf/meshes/camera_chin_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.498039215686275 0.498039215686275 0.498039215686275 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://sanbot_nano_urdf/meshes/camera_chin_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="camera_chin"
    type="fixed">
    <origin
      xyz="-0.0799999999999998 0 -0.1355"
      rpy="0 0 -1.5707963267949" />
    <parent
      link="head_link" />
    <child
      link="camera_chin_link" />
    <axis
      xyz="0 0 0" />
  </joint>

  <!--
  <transmission name="wheel_back_link_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wheel_back">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="wheel_back_link_motor">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wheel_left_link_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wheel_left">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="wheel_left_link_motor">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wheel_right_link_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wheel_right">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="wheel_right_link_motor">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>-->
  <transmission name="wing_left_link_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wing_left">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wing_left_link_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wing_right_link_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wing_right">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wing_right_link_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="neck_link_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="neck">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="neck_link_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="head_link_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="head">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="head_link_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <gazebo reference="wheel_back_link">
    <selfCollide>true</selfCollide>
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <gazebo reference="wheel_left_link">
    <selfCollide>true</selfCollide>
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <gazebo reference="wheel_right_link">
    <selfCollide>true</selfCollide>
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <gazebo reference="wing_left_link">
    <selfCollide>true</selfCollide>
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <gazebo reference="wing_right_link">
    <selfCollide>true</selfCollide>
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <gazebo reference="neck_link">
    <selfCollide>true</selfCollide>
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <gazebo reference="head_link">
    <selfCollide>true</selfCollide>
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <gazebo reference="base_link">
    <selfCollide>true</selfCollide>
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  <gazebo reference="body_link">
    <selfCollide>true</selfCollide>
    <turnGravityOff>true</turnGravityOff>
  </gazebo>
  
  <gazebo reference="neck">
    <provideFeedback>true</provideFeedback>
    <implicitSpringDamper>1</implicitSpringDamper>
    <implicitDampingFactor>0.1</implicitDampingFactor>
  </gazebo>
  
  <gazebo reference="head">
    <provideFeedback>true</provideFeedback>
    <implicitSpringDamper>1</implicitSpringDamper>
    <implicitDampingFactor>0.1</implicitDampingFactor>
  </gazebo>
  
  <gazebo reference="camera_chin_link">
    <sensor type="camera" name="camera_chin">
      <pose>0 0 0 -1.5708 1.5708 0</pose>
      <always_on>true</always_on>
      <update_rate>30</update_rate>
      <visualize>true</visualize>
      <camera>
        <image>
          <width>1280</width>
          <height>720</height>
          <format>R8G8B8</format>
        </image>
      </camera>
      <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
        <robotNamespace>/</robotNamespace>
        <cameraName>camera</cameraName>
        <imageTopicName>image_raw</imageTopicName>
        <cameraInfoTopicName>camera_info</cameraInfoTopicName>
        <frameName>camera_chin_link</frameName>
      </plugin>
    </sensor>
  </gazebo>

  <gazebo>
    <plugin name="control"
      filename="libgazebo_ros_control.so">
      <robotNamespace>/</robotNamespace>
    </plugin>
  </gazebo>

  <gazebo>
    <plugin name="imu_plugin" filename="libgazebo_ros_imu.so">
      <alwaysOn>true</alwaysOn>
      <bodyName>base_link</bodyName>
      <topicName>imu</topicName>
      <serviceName>imu_service</serviceName>
      <gaussianNoise>0.0</gaussianNoise>
      <updateRate>10.0</updateRate>
    </plugin>
  </gazebo>

  <gazebo>
    <plugin name="object_controller" filename="libgazebo_ros_planar_move.so">
      <commandTopic>cmd_vel</commandTopic>
      <odometryTopic></odometryTopic>
      <odometryFrame></odometryFrame>
      <odometryRate>0.0</odometryRate>
      <robotBaseFrame>base_link</robotBaseFrame>
    </plugin>
  </gazebo>
</robot>